src/js/coblocks-services.js

```diff
@@ -0,0 +1,15 @@
+( function() {
+	'use strict';
+	document.addEventListener( 'DOMContentLoaded', function() {
+		const servicesBlocks = document.querySelectorAll( '.wp-block-coblocks-service' );
+
+		for ( let i = 0; i < servicesBlocks.length; i++ ) {
+			const currentServiceBlock = servicesBlocks[ i ];
+			const hasImage = currentServiceBlock.getElementsByTagName( 'figure' ).length;
+
+			if ( 0 === currentServiceBlock.innerText.length && ! hasImage ) {
+				currentServiceBlock.remove();
+			}
+		}
+	} );
+}() );
```

Comment:
The way that the services block is written, there is always an empty service block in the InnerBlocks.Content on save. As far as I can tell, there is no way around this without re-writing the services block to not inject a new service when the previous one contains content.

This script loads on the front of site and removes all empty service blocks on the page.

Reply:
See my comment regarding adding this script as a dependency of the block.

Reply:
Although I'm not a fan of this we are migrating away from the services block so I'm ok with this existing until then since it resolves [WPEX-2415](https://godaddy-corp.atlassian.net/browse/WPEX-2415). It's not worth re-writing the block if we're removing it.

I would consider splitting this out into it's own PR so we can merge it independently if the 5 issues being resolved in this PR are not ready to merge.